.travis.yml

```diff
@@ -1,13 +1,23 @@
-sudo: required
+# Linting/Testing available at https://config.travis-ci.com/explore
+os: linux
+dist: xenial
 services:
   - postgresql
-  - redis-server
+  - redis
 addons:
   postgresql: "9.6"
 language: python
-matrix:
+env:
+  - INVOKE_NAUTOBOT_LOCAL=True
+jobs:
+  fast_finish: true
   include:
-    - python: "3.6"
+    - stage: Linting
+      python: "3.6"
+      script:
+        - "poetry run invoke tests --lint-only"
+    - stage: Unit Tests
+      python: "3.6"
```

Comment:
I love the concept here but this doesn't seem quite right, because even in the "Linting" stage, it still takes ~3 minutes to run, with ~90s being the Poetry environment bootstrap. (See: https://travis-ci.com/github/nautobot/nautobot/jobs/495740273)

Then, the "Unit Test" stage executes afterwards, bootstrapping Poetry again (90s), taking ~9 minutes to run in total. (See: https://travis-ci.com/github/nautobot/nautobot/jobs/495740274)

The net effect here is an _increase_ in total build time of 2-3 minutes when everything passes, which I don't think is what we want. We need some sort of caching between the stages so that Poetry doesn't have to be bootstrapped again. And if we could actually re-use the database that was created in the linting stage that would also save time.

My suggestion would just to revert the stages to matrix for now and just run all the tests with `--failfast`.

Reply:
Good catch, I had this in there in my testing.  I think we've talked about other ways of making this better using the docker images, I'll work on improving the build times more there.